core/blockchain.go

```diff
@@ -2119,6 +2163,9 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, er
 		if err != nil {
 			return it.index, err
 		}
+		if statedb.NoTrie() {
+			statedb.SetExpectedStateRoot(block.Root())
+		}
```

Comment:
The bellow line#2185 set state root too, duplicated?
statedb.SetExpectedStateRoot(block.Root())

Reply:
yes, it was a duplicate, i fixed